CSS module landing page (#23898)
mdn/content: files/en-us/web/css/compositing_and_blending/index.md

---

> title: Compositing and blending

Reviewer: ```suggestion
title: CSS compositing and blending
```
I realize that the spec does not have the beginning "CSS" but we can follow a consistent pattern on MDN and make it similar to titles for other modules. Not sure if we should let the slug be as is.

Author: i think we're going to redo the folder structure for CSS, so prefer not to redirect just to redirect again, though others may disagree.

  - Guide
  - Overview
  - Reference
spec-urls: 
  - https://drafts.fxtf.org/compositing/
  - https://www.w3.org/TR/compositing-1/
---

{{CSSRef}}

The **Compositing and Blending** CSS module defines how an element's background layers can be blended together, how an element can be blended with its container, and whether the element must create a new [stacking context](/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context).

Similar to blending effects available in many image editing applications, this module enables blending an element's background layers together and blending an element's content with that of its container. CSS can be used to define which blending mode should be used, if any, to blend an element's background images and colors into a single background image. It is also possible to define how an element's borders, background, and content, including text, emojis, and images, should be blended with the background of its container.

This module provides for 16 different blending modes.

> ### Example

Reviewer 2: Should this be H2? And why not show source for the example? And might we want to have more than one, and then call it Examples?

I would have the example after the Reference and Associated content, as we usually do for docs (interactive examples aside).

Author: I actually didn't want to have a header at all, but didn't know how to show EmbedLiveSample without a header ID. I want to keep the code hidden as this is supposed to be a very brief overview of "why" you would want to dig into the module, not an explanation of what the properties are or how they work. The example is part of the description, not an explanation, which is why it is with the description of the module.

  - Guide
  - Overview
  - Reference
spec-urls: 
  - https://drafts.fxtf.org/compositing/
  - https://www.w3.org/TR/compositing-1/
---

{{CSSRef}}

The **compositing and blending** CSS module defines how an element's background layers can be blended together, how an element can be blended with its container, and whether the element must create a new [stacking context](/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context).

The properties in this CSS module can be used to define the blending mode that should be used, if any, to blend an element's background images and colors into a single background image. This module provides 16 blending modes. You can also define how an element's borders, background, and content, including text, emojis, and images, should be blended with the background of its container.

### Compositing and blending in action

> In this example, each box has a border, two striped background images, and a solid color background. The common background for all the boxes contains a pattern of circles. The three boxes in the second row are set to blend with the background of the container. 

Author: ```suggestion
In this example, each box has a border, two striped background images, and a solid color background. The common background for all the boxes contains a pattern of circles. The three boxes in the second row are set to blend with the background of the container.
```

  - Guide
  - Overview
  - Reference
spec-urls: 
  - https://drafts.fxtf.org/compositing/
  - https://www.w3.org/TR/compositing-1/
---

{{CSSRef}}

> The **compositing and blending** CSS module defines how an element's background layers can be blended together, how an element can be blended with its container, and whether the element must create a new [stacking context](/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context).
> 
> Similar to blending effects available in many image editing applications, this module enables blending an element's background layers together and blending an element's content with that of its container. CSS can be used to define which blending mode should be used, if any, to blend an element's background images and colors into a single background image. It is also possible to define how an element's borders, background, and content, including text, emojis, and images, should be blended with the background of its container.
> 
> This module provides for 16 different blending modes.

Reviewer: ```suggestion
The **compositing and blending** CSS module defines how an element's background layers can be blended together, how an element can be blended with its container, and whether the element must create a new [stacking context](/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context).

The properties in this CSS module can be used to define the blending mode that should be used, if any, to blend an element's background images and colors into a single background image. This module provides 16 blending modes. You can also define how an element's borders, background, and content, including text, emojis, and images, should be blended with the background of its container.
```
deleted the sentence "Similar to.." because it conveyed the same info as the first sentence.

### Compositing and blending in action

> In this example, a border and two striped background images with a background color are provided for each box, with a background of a solid color with circles cut out set on the entire example.

Reviewer: ```suggestion
In this example, each box has a border, two striped background images, and a solid color background. The common background for all the boxes contains a pattern of circles. The three boxes in the second row are set to blend with the background of the container. 
```

    repeating-linear-gradient(-45deg, blue 0 15px, lightblue 15px 30px);
  background-size: 150px 150px;
  background-repeat: no-repeat;
  background-position: top left, bottom right;
  background-color: palegoldenrod;
  text-align: center;
  padding-top: 150px;
  font-family: sans-serif;
  box-sizing: border-box;
  border: 5px solid black;
}
div:nth-of-type(3n+1){
  background-blend-mode: normal;
}
div:nth-of-type(3n+2){
  background-blend-mode: multiply;
}
div:nth-of-type(3n+3){
  background-blend-mode: overlay;
}
div:nth-of-type(n + 4) {
  mix-blend-mode: difference;
}
/* put a pink background with transparent round holes that covers the entire element, and lay the examples in two rows with three columns each */
section {
  padding: 0.75em;
  background: radial-gradient(circle, transparent 0 20px, rgb(255, 200, 200) 20px); 
  background-size: 60px 60px; 
  background-position: center;
  display: inline-grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 1em;
}
/* make some of the text more legible */
span {
  background-color: #ffffff99;
}
```

{{ EmbedLiveSample('Compositing_and_blending_in_action', "630", "300") }}

>  The last three boxes are set to blend with the background of the container. Note how the background, border, and the content are all impacted. To see the code for this, [view the source on Github](https://github.com/mdn/content/blob/main/files/en-us/web/css/compositing_and_blending/index.md?plain=1).

Reviewer: ```suggestion
 Notice how the background, border, and the content are all impacted as a result of the blending. To see the code for this sample, [view the source on Github](https://github.com/mdn/content/blob/main/files/en-us/web/css/compositing_and_blending/index.md?plain=1).
```

- {{cssxref("isolation")}}
- {{cssxref("mix-blend-mode")}}

## Related concepts

> - {{cssxref("blend-mode")}} data type
> 
> - {{cssxref("backdrop-filter")}} CSS property
> - {{cssxref("filter")}} CSS property
> - {{cssxref("mask-composite")}} CSS property
> - {{cssxref("background-color")}} CSS property
> - {{cssxref("background-image")}} CSS property
> 
> - {{glossary("stacking context")}} glossary term

Author: ```suggestion
- {{cssxref("blend-mode")}} data type

- {{cssxref("backdrop-filter")}} CSS property
- {{cssxref("filter")}} CSS property
- {{cssxref("mask-composite")}} CSS property
- {{cssxref("background-color")}} CSS property
- {{cssxref("background-image")}} CSS property

- {{glossary("stacking context")}} glossary term
```

- {{cssxref("isolation")}}
- {{cssxref("mix-blend-mode")}}

## Related concepts

- {{cssxref("blend-mode")}} data type
- {{cssxref("filter")}} CSS property
- {{cssxref("backdrop-filter")}} CSS property
- {{cssxref("mask-composite")}} CSS property
- {{cssxref("background-color")}} CSS property
- {{cssxref("background-image")}} CSS property
- {{glossary("stacking context")}} glossary term
- {{ SVGElement("feBlend")}} SVG filter primitive

> - {{ SVGElement("feComposite")}} SVG filter primitive

Author: I set them in order of relevance. Filter, which does something similar, seemed more relevant than background properties, which I then did put in alphabetical order.

## Specifications

{{Specifications}}

## See also

> - [CSS filter effects](/en-US/docs/Web/CSS/Filter_Effects) module enables applying filter effects to images like blurring and changing color intensity.

Author: ```suggestion
- Properties in the [CSS filter effects](/en-US/docs/Web/CSS/Filter_Effects) module enable applying filters effects, such as blurring and changing color intensity, to images, backgrounds, and borders.
```
